### ISSUE TITLE
Dual purpose SERVER_NAME

### ISSUE THREAD

**Issue by sgleske-ias** [SergK/nexus3-ssl#3]
# Question

@SergK (I'm also @samrocketman) do you mind if I propose a change which dual purposes the domain `SERVER_NAME` to also be used as a docker registry?

# Background

This would effectively get rid of `DOCKER_DEV_NAME`, `DOCKER_VIRTUAL_NAME`, and `UPSTREAM_DOCKER_REPO` [from `env.config`][1].

Example nginx config:
```
upstream nexus {
  server nexus:8081;
}
upstream docker-hosted {
  #port of the hosted docker registry
  server nexus:8082;
}
...
server {
  ...
  location / {
    #proxypass docker registry if docker client is used
    if ($http_user_agent ~ docker) {
      proxy_pass http://docker-hosted;
    }
    #fall back to the web browser (because first proxy_pass wins)
    proxy_pass http://nexus;
    ...
  }
}
```

Let's say for example, your nexus frontend is hosted at `https://nexus.sandbox.example.com/` (per the `env.config` `SERVER_NAME`).  The above pseudo code `nginx.config.tpl` would essentially allow the Nexus frontend to be visited by a browser but also allow docker commands with it.  It would be pretty seamless.

    docker login nexus.sandbox.example.com
    docker push nexus.sandbox.example.com/my_docker_image:latest

I got the idea from the [Nexus blog on configuring nginx with Nexus][2].

[1]: https://github.com/SergK/nexus3-ssl/blob/fc9694fc4143ef7eba7cd5faac5f4f234da02d1f/env.config
[2]: http://www.sonatype.org/nexus/2017/03/02/setting-up-a-docker-private-registry-with-authentication-using-nexus-and-nginx/#running-nginx-as-reverse-proxy-for-nexus

**Comment by SergK** [comment]
Hi @sgleske-ias ! Thank you for the feedback. Just to clarify, the original idea of master branch was to provide different endpoints for images in **dev** and **production**, so the image with prefix

- **docker-dev.sandbox.example.com/some_image:tag**  comes from development pipelines

-  **docker-prod.sandbox.example.com/some_image:tag** comes from production pipelines

So you are free to fork and create separate branch, e.g. I did this in **custom_deploy**  branch https://github.com/SergK/nexus3-ssl/tree/custom_deploy You can compare it with master and they are completely different

To summarize, I just want to leave master branch as it was originally designed for, but will accept changes regarding your idea in new branch.

**Comment by sgleske-ias** [comment]
Ok, sounds good.
